### DISCUSSION TITLE
Unexpected/unreported bond configuration changes by SDMolSupplier

### DISCUSSION THREAD

**Discussion by someonetookmyname** [rdkit/rdkit#6497] [Q&A] [answered]
Hi,

I'm writing to inquire about the behavior of SDMolSupplier towards removing and/or moving bond configurations during SDF import.

Some erroneous/nonsensical bond configurations are being removed during import without any alert/announcement.  For example, wedge/hash bonds originating at non-stereocenters.

In addition, some legitimate (albeit less desirable) bond configurations are getting moved without any alert/announcement.  For example, wedge/hash bonds in rings, wedge/hash bonds between two stereocenters.

In all cases, I'm using sanitize=False, removeHs=False, strictParsing=False:

```
w = Chem.SDWriter(‘out.sdf’)
w.SetForceV3000(True)
mols = Chem.SDMolSupplier('in.sdf', sanitize=False, removeHs=False, strictParsing=False)
for mol in mols:
    w.write(mol)
```

Is there a way to prevent these removals/changes and/or have specific alerts/announcements that they have taken place?

Many constitute user-drawing errors, which we would like to actively note/catch during processing.

I have attached in.sdf (as in.txt to allow attachment here) and out.sdf (as out.txt) and also a pdf overview of the changes.

Thank you for any guidance.

Best, Andrew
[configurationchanges.pdf](https://github.com/rdkit/rdkit/files/11878403/configurationchanges.pdf)
[in.txt](https://github.com/rdkit/rdkit/files/11878408/in.txt)
[out.txt](https://github.com/rdkit/rdkit/files/11878409/out.txt)

**Comment by greglandrum** [comment, marked as answer]
Hi @someonetookmyname, although the RDKit removes the bond wedges by default, it does save information about where they were. You can re-apply the bond wedges using `Chem.ReapplyMolBlockWedging()`:
```
In [4]: m = Chem.MolFromMolBlock('''
   ...:   Mrv2305 06302310332D          
   ...: 
   ...:   0  0  0     0  0            999 V3000
   ...: M  V30 BEGIN CTAB
   ...: M  V30 COUNTS 4 3 0 0 0
   ...: M  V30 BEGIN ATOM
   ...: M  V30 1 C -4.75 3.3333 0 0
   ...: M  V30 2 C -3.4163 4.1033 0 0 CFG=2
   ...: M  V30 3 C -2.0826 3.3333 0 0
   ...: M  V30 4 C -3.4163 5.6433 0 0
   ...: M  V30 END ATOM
   ...: M  V30 BEGIN BOND
   ...: M  V30 1 1 1 2
   ...: M  V30 2 1 2 3
   ...: M  V30 3 1 2 4 CFG=1
   ...: M  V30 END BOND
   ...: M  V30 END CTAB
   ...: M  END
   ...: ''')

In [5]: Chem.ReapplyMolBlockWedging(m)

In [6]: m.Debug()
Atoms:
        0 6 C chg: 0  deg: 1 exp: 1 imp: 3 hyb: 4 arom?: 0 chi: 0
        1 6 C chg: 0  deg: 3 exp: 3 imp: 1 hyb: 4 arom?: 0 chi: 0
        2 6 C chg: 0  deg: 1 exp: 1 imp: 3 hyb: 4 arom?: 0 chi: 0
        3 6 C chg: 0  deg: 1 exp: 1 imp: 3 hyb: 4 arom?: 0 chi: 0
Bonds:
        0 0->1 order: 1 conj?: 0 aromatic?: 0
        1 1->2 order: 1 conj?: 0 aromatic?: 0
        2 1->3 order: 1 dir: 1 conj?: 0 aromatic?: 0
```

**Reply by someonetookmyname** [reply]
Thanks so much, Greg!  Much appreciated .. just what I was looking for... smile
